applications/StructuralMechanicsApplication/custom_utilities/large_displacement_variables.h

```diff
@@ -0,0 +1,157 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    
+//
+
+#if !defined(KRATOS_LARGE_DISPLACEMENT_VARIABLES_H_INCLUDED )
+#define  KRATOS_LARGE_DISPLACEMENT_VARIABLES_H_INCLUDED
+
+// System includes
+
+// External include
+
+// Project includes
+#include "includes/define.h"
+#include "includes/node.h"
+#include "geometries/geometry.h"
+#include "utilities/matrix_vector_utilities.h"
+
+namespace Kratos
+{
+
+/**
+ * @class LargeDisplacementDifferentialVariables
+ * @ingroup StructuralMechanicsApplication
+ * @brief Differential variables for large displacements with reference configuration.
+ * @details Provides deformation gradient, shape functions gradients, and
+ * Jacobian information for a geometry relative to its reference configuration.
+ */
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) LargeDisplacementDifferentialVariables
+{
+public:
+    typedef Geometry<Node<3>> GeometryType;
+
+    explicit LargeDisplacementDifferentialVariables(GeometryType const& rGeom);
+
+    /// Deformation gradient.
+    const Matrix& F(std::size_t IntegrationIndex, bool IsAxisymmetric = false);
```

Comment:
I would make this method const, since i believe it should not compute nothing and instead just give bach the value that is already computed

Reply:
It changes a member variable. Only if we change the class behavior or declare the variable mutable can we do this. Let's leave this until we agree about the class design.